### DISCUSSION TITLE
Sync Information on Client Side

### DISCUSSION THREAD

**Discussion by hsharma89** [Mimetis/Dotmim.Sync#928] [Q&A]
Hi,
Greetings of the day.
I have a requirement to show database sync status to my clients, like whether the database is 100% synced. I am using multi-tenant database for my application. Is there any functionality available in Dotmim to achieve this feature?

**Comment by Mimetis** [comment]
You have the `IProgress<ProgressArgs>` available.
More info here : https://dotmimsync.readthedocs.io/Progression.html#id1

**Reply by hsharma89** [reply]
Thanks a lot for your quick response. We have already implemented this on our desktop application, This shows the sync percentage when we hit sync. Is there any way we can check if the database is fully synced without syncing the data? Say we have an indicator on the application which tells that Database is fully synced or there are some unsynced data on the server/client.

**Reply by Mimetis** [reply]
Oh, I got it !

You can use the [GetEstimatedChangesCountAsync](https://dotmimsync.readthedocs.io/Orchestrators.html?#getestimatedchangescountasync) method on both orchestrator, that will make an estimation of what is not synced, yet.
It's an estimation, since it is not resolving any conflicts that may potentially occurs.
This method works only if you have already made a first sync of your client database, as it relies on DMS metadata (tracking tables, stored procedures ...)

Here is a quick example:

``` cs
var agent = new SyncAgent(clientProvider, serverProvider, options);

var cScopeInfoClient = await agent.LocalOrchestrator.GetScopeInfoClientAsync(scopeName, parameters);
var clientEstimatedChanges = await agent.LocalOrchestrator.GetEstimatedChangesCountAsync(cScopeInfoClient);
var serverEstimatedChanges = await agent.RemoteOrchestrator.GetEstimatedChangesCountAsync(cScopeInfoClient);

Debug.WriteLine($"Total Changes From Client to upload:{clientEstimatedChanges.ClientChangesSelected.TotalChangesSelected}");
Debug.WriteLine($"Total Changes From Server to download:{serverEstimatedChanges.ServerChangesSelected.TotalChangesSelected}");

foreach (var estimation in clientEstimatedChanges.ClientChangesSelected.TableChangesSelected)
{
    Debug.WriteLine($"Client Table {estimation.TableName}: Updates/Inserts:{estimation.Upserts}");
    Debug.WriteLine($"Client Table {estimation.TableName}: Deletes:{estimation.Deletes}");
}

foreach (var estimation in serverEstimatedChanges.ServerChangesSelected.TableChangesSelected)
{
    Debug.WriteLine($"Server Table {estimation.TableName}: Updates/Inserts:{estimation.Upserts}");
    Debug.WriteLine($"Server Table {estimation.TableName}: Deletes:{estimation.Deletes}");
}

```

``` bash
Total Changes From Client to upload:4
Total Changes From Server to download:2

Client Table ProductCategory: Updates/Inserts:2
Client Table ProductCategory: Deletes:0
Client Table Product: Updates/Inserts:2
Client Table Product: Deletes:0

Server Table SalesOrders: Updates/Inserts:2
Server Table SalesOrders: Deletes:0
```

**Reply by hsharma89** [reply]
Thanks a lot. I'll let you know once done.
